File: src/data_process/VideoProcessor.cpp
// ============================
// VideoProcessor Source File
// Process video data into luma
// ============================


// Includes
#include "VideoProcessor.h"


// VideoProcessor constructor
VideoProcessor::VideoProcessor(const cv::String& name)
{

    cv::String video_path = "/home/riley/Projects/riley/data/videos/" + name + ".mp4";

    cv::VideoCapture capture(video_path);

    cv::Mat frame;
    bool okay;
    okay = capture.read(frame);

    int frame_count = 0;

    while (okay) {
        frame_count++;

        std::cout << "Frame: " << frame_count << std::endl;

        for (int i = 0; i < frame.rows; i++) {
            const cv::Vec3b *Mi = frame.ptr<cv::Vec3b>(i);

            for (int j = 0; j < frame.cols; j++) {
                cv::Vec3b pixel = Mi[j];

                double luma = 0.2126 * pixel[0] + 0.7152 * pixel[1] + 0.0722 * pixel[2];

//                std::cout << "Luma of "
//                          << "[" << frame_count << ", " << i << ", " << j << "]: "
//                          << luma << std::endl;
            }
        }

        okay = capture.read(frame);

    }

}

File: src/data_process/src/VideoProcessor.py
# venv/bin/python3

import cv2

name = input("Video name: ")
video_path = "/home/riley/Projects/riley/data/videos/" + name + ".mp4"
data_path = "/home/riley/Projects/riley/data/store/" + name + ".txt"

cap = cv2.VideoCapture(video_path)

okay, frame = cap.read()
frame_number = 0
print(f'Video appears to be {len(frame[0])} x {len(frame)}')


# Process each frame
while okay and frame_number < 20:
    frame_number += 1
    data = ""

    # Print frame number to keep updated on time to completion
    print(f"Frame number: {frame_number}")

    for row in frame:
        for pixel in row:
            luma = round(0.2126*pixel[0] + 0.7152*pixel[1] + 0.0722*pixel[2], 2)
            data += f"{luma},"
            # print(f'{frame_number}: {pixel[0]}, {pixel[1]}, {pixel[2]}')

    # Remove the last comma and add a new line
    data = data[:-1] + "\n"
    with open(data_path, "a") as file:
        file.write(data)

    okay, frame = cap.read()
